src/Emulator/Extensions/Mocks/HPSHostController.cs

```diff
@@ -171,6 +174,92 @@ private byte[] GetBytesFromSlave(int count, TimeInterval timeInterval)
             return result;
         }
 
+        // When we need to read for more than 255 bytes, we will need this method
+        private byte[] GetLongBytes(int count, TimeInterval timeInterval) {
+            if (count <= 255) {
+                return GetBytesFromSlave(count, timeInterval);
+            }
+            var result = new byte[count];
+            int rounds = count /255;
+            int remainder = count % 255; 
+            for (int i = 0; i <= rounds; i ++) {
+                int toRead = i == rounds ? remainder : 255;
+                Array.Copy(GetBytesFromSlave(toRead, timeInterval), 0, result, i * 255, toRead);
+            }
+            return result;
+        }
+
+        public void StartSocketServer(TimeInterval timeInterval)
+        {  
+            Console.WriteLine(currentSlave.GetType().Name);
+            string path = "/tmp/i2c.sock";
+            bool needError = false;
+            if (File.Exists(path))
+            {
+                File.Delete(path);
+            }
+            var socket = new Socket(AddressFamily.Unix, SocketType.Stream, ProtocolType.Unspecified);
+            socket.Bind(new UnixEndPoint(path));
+            socket.Listen(0);
+            Console.WriteLine("Server started waiting for client to connect...");
+            Console.WriteLine("Starting on: "+ path);
+            currentSlave.Read(0);
+            var connection = socket.Accept();
+            Console.WriteLine("Connection established");
+            // Doubleword();
+
+            // byte[] commandBuffer = new byte[500];
+            while(SocketConnected(connection)){
+                var commandBuffer = new byte[1];
+                var lengthBytes = new byte[2];
+                int length;
+                try {
+                    connection.Receive(commandBuffer, 0, commandBuffer.Length, SocketFlags.None);
+                }
+                catch(Exception e)
+                {
+                    Console.WriteLine("{0} Exception caught.", e);
+                    continue;
+                }
+                switch(commandBuffer[0])
+                {
+                    case (byte)I2CMessageType.Stop:
```

Comment:
What's the purpose of this command? If you want to terminate the socket maybe it should do `break` instead of `continue`?

Reply:
Stop and start are particular signals on the I2C bus that indicate the start and end of transaction. We don't want to terminate the socket when a stop is received, because then we wouldn't be able to respond to subsequent requests.

Reply:
Yup exactly what David said